index.js

```diff
@@ -359,10 +268,10 @@
   };
 
   //  $$type :: a -> String
-  var $$type = function(x) {
-    return x != null && toString.call(x['@@type']) === '[object String]' ?
+  var $$type = function $$type(x) {
+    return x != null && $$type(x['@@type']) === 'String' ?
```

Comment:
`$$type({'@@type': {'@@type': 'String'}}) === {'@@type': 'String'}` which invalidates the type signature. Not sure if it really matters though.